type-definitions/immutable.d.ts

```diff
@@ -5160,12 +5160,39 @@ declare namespace Immutable {
    */
   function fromJS(
     jsValue: unknown,
-    reviver?: (
+    reviver: (
```

Comment:
No way this is correct - I never had problems skipping this argument in runtime.

Reply:
hi @ntucker . That's an overload for the definition right after. It does say : 
- if the reviver method is set, then the return type is `Collection<unknown, unknown>`
- if the reviver is not set of undefined, then we can calculate the return type

Reply:
https://github.com/immutable-js/immutable-js/issues/1935

https://stackblitz.com/edit/typescript-sk2pyy?file=index.ts

Reply:
This may very well be a bug with typescript, but I'm fairly certain you'd still be interested in supporting typescript versions less than 5.1

Reply:
Let's continue in #1935 to avoid multiplying conversation in different threads